packages/gasket-plugin-command/lib/index.d.ts

```diff
@@ -0,0 +1,109 @@
+import type { MaybeAsync, GasketConfig, Hook } from '@gasket/core';
+import type { PluginData } from '@gasket/plugin-metadata';
+import type { Command, Option } from 'commander';
+
+export interface GasketCommandDefinition {
+  /* Command id/name */
+  id: string;
+
+  /* Command description */
+  description: string;
+
+  /* Command arguments */
+  args?: Array<GasketArgDefinition>;
```

Comment:
Any thoughts on this being an object instead? Personally I would find this more ergonomic:

```typescript
```suggestion
  args?: Record<string, GasketArgDefinition>;
```

...with `name` being implied by the key of the object. It should make it easier to "index" the arguments visually. I'd rather write:

```
args: {
  username: { },
  password: { }
}
```

...over:

```
args: [
  { name: 'username', ... },
  { name: 'password', ... }
]
```

Reply:
It easily could be..... @agerard-godaddy thoughts?

Reply:
Variadic options would be a little weird. If it's agreed I'll likely create a follow-up ticket.

```
args: {
  'no-prompt': { }
}
```

Reply:
👍 I like the object suggestion

Reply:
Ticket -> https://godaddy-corp.atlassian.net/browse/PFX-621?atlOrigin=eyJpIjoiYWQ4ZDU3OTBmMzEyNDE5MjhlYjM2M2E5MDdmNWUxNTIiLCJwIjoiaiJ9